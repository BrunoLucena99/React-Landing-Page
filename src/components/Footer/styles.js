import styled from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  text-align: center;
  border-top: ${({ theme }) => `0.1rem solid ${theme.colors.mediumGray}`};

  a {
    color: inherit;
    text-decoration: none;
  }

  ${TextComponent} {
    font-size: ${({ theme }) => theme.font.sizes.small};
  }

  ${SectionContainer} {
    padding-top: 0;
    padding-bottom: 0;
  }
`;
