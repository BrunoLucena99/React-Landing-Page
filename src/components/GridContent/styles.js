import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const HTMLContainer = styled.div`
  margin: ${({ theme }) => theme.spacings.xhuge} 0;
`;
