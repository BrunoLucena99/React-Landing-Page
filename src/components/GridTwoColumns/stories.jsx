import GridTwoColumns from '.';
import mock from '../../mocks/GridTwoColumns';

export default {
  title: 'GridTwoColumns',
  component: GridTwoColumns,
  args: mock,
  argTypes: {
    text: { type: 'string' },
    title: { type: 'string' },
    image: { type: 'string' },
    hasBackground: { type: 'boolean' },
  },
};

export const Template = (args) => <GridTwoColumns {...args} />;
