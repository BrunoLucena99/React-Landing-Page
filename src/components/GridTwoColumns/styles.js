import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
