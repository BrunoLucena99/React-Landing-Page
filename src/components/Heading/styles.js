import styled, { css } from 'styled-components';

const titleSizes = {
  small: ({ font }) => css`
    font-size: ${font.sizes.medium};
  `,
  medium: ({ font }) => css`
    font-size: ${font.sizes.large};
  `,
  big: ({ font }) => css`
    font-size: ${font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSizes[size](theme)};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}
`;
