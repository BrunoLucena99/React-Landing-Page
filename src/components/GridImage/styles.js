import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
