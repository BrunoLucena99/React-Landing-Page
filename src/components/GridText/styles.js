import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  counter-reset: grid-counter;
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  ${HeadingTitle} {
    position: relative;
    left: 5rem;
  }

  ${HeadingTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;
