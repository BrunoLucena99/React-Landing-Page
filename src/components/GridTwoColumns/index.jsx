import P from 'prop-types';
import { Container, TextContainer, ImageContainer, Image } from './styles';
import SectionBackground from '../SectionBackground';
import Heading from '../Heading';
import TextComponent from '../TextComponent';

const GridTwoColumns = ({ title, text, image, hasBackground = false }) => (
  <SectionBackground hasBackground={hasBackground}>
    <Container>
      <TextContainer>
        <Heading as="h2" colorDark={!hasBackground} uppercase>
          {title}
        </Heading>
        <TextComponent>{text}</TextComponent>
      </TextContainer>
      <ImageContainer>
        <Image src={image} />
      </ImageContainer>
    </Container>
  </SectionBackground>
);

GridTwoColumns.propTypes = {
  title: P.string.isRequired,
  text: P.string.isRequired,
  image: P.string.isRequired,
  hasBackground: P.bool,
};

export default GridTwoColumns;
