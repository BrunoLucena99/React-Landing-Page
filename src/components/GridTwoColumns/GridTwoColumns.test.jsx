import { renderTheme } from '../../styles/RenderTheme';
import GridTwoColumns from '.';
import { screen } from '@testing-library/react';
import mock from '../../mocks/GridTwoColumns';

describe('<GridTwoColumns />', () => {
  it('should render correctly with default values', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should correctly when background is setted', () => {
    const { container } = renderTheme(
      <GridTwoColumns {...mock} hasBackground />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should image showing correctly', () => {
    renderTheme(<GridTwoColumns {...mock} />);
    expect(screen.getByRole('img', { src: mock.image })).toBeInTheDocument();
  });
});
