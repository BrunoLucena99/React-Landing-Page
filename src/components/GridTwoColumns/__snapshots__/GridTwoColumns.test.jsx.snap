// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should correctly when background is setted 1`] = `
.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4.0rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  background-color: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4"
        >
          Grid two columns title
        </h1>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
    Nobis adipisci earum accusantium architecto.
    Quo minus odio cum vel, ipsum cumque omnis.
    Obcaecati commodi repellendus fugiat quos est.
    A, similique saepe?
        </p>
      </div>
      <div
        class=""
      >
        <img
          class=""
          src="assets/images/js.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridTwoColumns /> should render correctly with default values 1`] = `
.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4.0rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4"
        >
          Grid two columns title
        </h1>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
    Nobis adipisci earum accusantium architecto.
    Quo minus odio cum vel, ipsum cumque omnis.
    Obcaecati commodi repellendus fugiat quos est.
    A, similique saepe?
        </p>
      </div>
      <div
        class=""
      >
        <img
          class=""
          src="assets/images/js.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
